### PR TITLE
added home view

### PR DESCRIPTION
**Pull request by Pelozo**
## Summary:

- Added home screen with news list and profile fragments

## Trello card

- [[U4] Home Page: Tabs de navegación](https://trello.com/c/WpLAqkAJ)

## Preview

Untitled video ‐ Made with Clipchamp

### CODE REVIEW

File: WoloxAndroidBootstrap/app/src/main/java/ar/com/wolox/android/example/ui/home/news/list/adapter/NewsAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.apply {
+            title.text = news[position].commenter
+            description.text = news[position].comment
+            image.glideload(news[position].avatar)
+            news[position].createdAt.toPrettyDate()?.let {
+                time.text = it
+            }
+            liked.apply {
```
Comment (jcruz-wolox) [inline WoloxAndroidBootstrap/app/src/main/java/ar/com/wolox/android/example/ui/home/news/list/adapter/NewsAdapter.kt]: You can use a selector to solve this logic

File: WoloxAndroidBootstrap/app/src/main/res/layout/fragment_add_new.xml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            android:id="@+id/h40Guide"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:orientation="horizontal"
+            app:layout_constraintGuide_percent="0.4"/>
+
+        <EditText
+            android:id="@+id/titleInput"
+            android:layout_width="match_parent"
+            android:layout_height="56dp"
```
Comment (jcruz-wolox) [inline WoloxAndroidBootstrap/app/src/main/res/layout/fragment_add_new.xml]: Remove hardcode dimen

File: WoloxAndroidBootstrap/app/src/main/res/layout/fragment_add_new.xml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            android:inputType="text"
+            android:maxLines="1"
+            app:layout_constraintTop_toBottomOf="@id/h40Guide"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent" />
+
+        <EditText
+            android:id="@+id/descriptionInput"
+            android:layout_width="match_parent"
+            android:layout_height="56dp"
```
Comment (jcruz-wolox) [inline WoloxAndroidBootstrap/app/src/main/res/layout/fragment_add_new.xml]: Rompe hardcode dime

File: WoloxAndroidBootstrap/app/src/main/res/layout/fragment_home.xml [hunk 4]
```diff
@@ -8,14 +8,36 @@
...
-            android:text="TODO"/>
+            app:title="Wolox"
+            app:navigationIcon="@drawable/wolox_logo"
+            android:paddingLeft="@dimen/spacing_large"
+            android:paddingRight="@dimen/spacing_large"/>
+
+        <com.google.android.material.tabs.TabLayout
+            android:id="@+id/tabLayout"
+            android:layout_width="0dp"
+            android:layout_height="48dp"
```
Comment (jcruz-wolox) [inline WoloxAndroidBootstrap/app/src/main/res/layout/fragment_home.xml]: Remove hardcode dimen

File: WoloxAndroidBootstrap/app/src/main/res/layout/fragment_news.xml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintRight_toRightOf="parent"
+            android:layout_marginRight="@dimen/spacing_large"
+            android:layout_marginBottom="@dimen/spacing_large"
+            android:src="@android:drawable/ic_input_add"
+            app:tint="@android:color/white"
+            android:alpha="0.7"
+            app:maxImageSize="32dp"
```
Comment (jcruz-wolox) [inline WoloxAndroidBootstrap/app/src/main/res/layout/fragment_news.xml]: Remove hardcode dimen

File: WoloxAndroidBootstrap/app/src/main/res/layout/fragment_profile.xml [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+        <androidx.constraintlayout.widget.Guideline
+            android:id="@+id/h40Guide"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:orientation="horizontal"
+            app:layout_constraintGuide_percent="0.4"/>
+
+        <com.google.android.material.imageview.ShapeableImageView
+            app:shapeAppearanceOverlay="@style/roundedImageViewRounded"
+            android:layout_width="72dp"
```
Comment (jcruz-wolox) [inline WoloxAndroidBootstrap/app/src/main/res/layout/fragment_profile.xml]: Remove hardcode dimen

File: WoloxAndroidBootstrap/app/src/main/res/layout/fragment_profile.xml [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+            android:layout_height="72dp"
```
Comment (jcruz-wolox) [inline WoloxAndroidBootstrap/app/src/main/res/layout/fragment_profile.xml]: Here too

File: WoloxAndroidBootstrap/app/src/main/res/layout/fragment_profile.xml [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+            app:srcCompat="@drawable/mock_avatar"
+            app:layout_constraintStart_toStartOf="@id/coverImage"
+            app:layout_constraintTop_toBottomOf="@id/coverImage"
+            app:layout_constraintBottom_toBottomOf="@id/coverImage"
+            android:layout_marginLeft="@dimen/static_spacing_medium_extra"/>
+
+        <Space
+            android:id="@+id/space"
+            android:layout_width="wrap_content"
+            android:layout_height="48dp"
```
Comment (jcruz-wolox) [inline WoloxAndroidBootstrap/app/src/main/res/layout/fragment_profile.xml]: Here too

File: WoloxAndroidBootstrap/app/src/main/res/layout/fragment_profile.xml [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+
+        <TextView
+            android:id="@+id/nameText"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            app:layout_constraintTop_toBottomOf="@+id/coverImage"
+            app:layout_constraintStart_toStartOf="parent"
+            android:layout_marginLeft="@dimen/static_spacing_largest_extra"
+            android:text="Mario Trusso"
+            android:textSize="18sp"/>
```
Comment (jcruz-wolox) [inline WoloxAndroidBootstrap/app/src/main/res/layout/fragment_profile.xml]: Here too

File: WoloxAndroidBootstrap/app/src/main/res/layout/fragment_profile.xml [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+        <TextView
+            android:id="@+id/locationText"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            app:layout_constraintTop_toBottomOf="@+id/nameText"
+            app:layout_constraintBottom_toBottomOf="@id/space"
+            app:layout_constraintStart_toEndOf="@id/locationImage"
+            android:layout_marginLeft="@dimen/spacing_medium"
+            android:text="Milano, Italy"
+            android:textSize="11sp"/>
```
Comment (jcruz-wolox) [inline WoloxAndroidBootstrap/app/src/main/res/layout/fragment_profile.xml]: Here too

File: WoloxAndroidBootstrap/app/src/main/res/layout/fragment_home.xml [hunk 11]
```diff
@@ -8,14 +8,36 @@
...
+        <androidx.appcompat.widget.Toolbar
+            android:id="@+id/toolbar"
+            android:layout_width="match_parent"
+            android:layout_height="?attr/actionBarSize"
             app:layout_constraintTop_toTopOf="parent"
-            app:layout_constraintBottom_toBottomOf="parent"
             app:layout_constraintStart_toStartOf="parent"
             app:layout_constraintEnd_toEndOf="parent"
-            android:text="TODO"/>
+            app:title="Wolox"
```
Comment (jcruz-wolox) [inline WoloxAndroidBootstrap/app/src/main/res/layout/fragment_home.xml]: Remove hardcode string

File: WoloxAndroidBootstrap/app/src/main/res/layout/item_new.xml [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools">
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="@dimen/row_height">
+
+        <ImageView
+            android:id="@+id/newImage"
+            android:layout_width="48dp"
```
Comment (jcruz-wolox) [inline WoloxAndroidBootstrap/app/src/main/res/layout/item_new.xml]: Remove hardcode dimen

File: WoloxAndroidBootstrap/app/src/main/res/layout/item_new.xml [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+            android:layout_height="48dp"
```
Comment (jcruz-wolox) [inline WoloxAndroidBootstrap/app/src/main/res/layout/item_new.xml]: Here too

File: WoloxAndroidBootstrap/app/src/main/res/layout/item_new.xml [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginLeft="@dimen/static_spacing_largest"
+            android:layout_marginRight="@dimen/static_spacing_largest"
+            android:textColor="@android:color/black"
+            app:layout_constraintBottom_toTopOf="@+id/descriptionText"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent"
+            android:textSize="18sp"
```
Comment (jcruz-wolox) [inline WoloxAndroidBootstrap/app/src/main/res/layout/item_new.xml]: Remove hardcode dimen

**Review by jcruz-wolox — CHANGES_REQUESTED** [review]
I have any comments:

- Put texts and dimensions in dimens.xml and strings.xml as appropriate.
- The like/dislike print logic can be resolve using selector

**Review by jcruz-wolox — APPROVED** [review]
Approved for me.

File: WoloxAndroidBootstrap/app/src/main/java/ar/com/wolox/android/example/ui/home/HomeFragment.kt [hunk 15]
```diff
@@ -2,13 +2,41 @@ package ar.com.wolox.android.example.ui.home
...
+
+    private fun setTabLayout() {
+        binding.viewPager.adapter = SimpleFragmentPagerAdapter(childFragmentManager).apply {
+            addFragments(
+                    newsFragment to getString(R.string.news_toolbar_title),
+                    profileFragment.get() to getString(R.string.profile_toolbar_title)
+            )
+        }
+        binding.tabLayout.apply {
+            binding.tabLayout.setupWithViewPager(binding.viewPager)
```
Comment (ffasitella) [inline WoloxAndroidBootstrap/app/src/main/java/ar/com/wolox/android/example/ui/home/HomeFragment.kt]: Just call `setupWithViewPager(viewPager)`

File: WoloxAndroidBootstrap/app/src/main/java/ar/com/wolox/android/example/utils/extensions/StringExtensionFunctions.kt [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+
+fun String.isEmailValid(): Boolean {
+    return !TextUtils.isEmpty(this) && Patterns.EMAIL_ADDRESS.matcher(this).matches()
+}
+
+fun String.toPrettyDate(): String? {
+    return try {
+        PrettyTime()
+            .format(Date(
+                SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'").parse(this).time
```
Comment (ffasitella) [inline WoloxAndroidBootstrap/app/src/main/java/ar/com/wolox/android/example/utils/extensions/StringExtensionFunctions.kt]: Move this string to a constant

**Review by ffasitella — CHANGES_REQUESTED** [review]
Niiice! Just left two style comments

**Review by ffasitella — APPROVED** [review]

**Pull request merged**